scertecd/scertecd.go

```diff
@@ -0,0 +1,814 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// The scertecd package provides the code that fetches new TLS certs
+// from LetsEncrypt as needed and puts them in setec before they
+// expire. The code can run either in the foreground once, or most
+// commonly as an HTTP server daemon.
+//
+// It populates the following setec keys:
+//
+//   - {prod,dev}/certd/acme-key: the private key for the ACME account, as a PEM-encoded ECDSA key
+//   - {prod,dev}/certd/https-cert/{domain-name}/rsa: PEM of private key, domain cert, LetsEncrypt cert
+//   - {prod,dev}/certd/https-cert/{domain-name}/ecdsa: PEM of private key, domain cert, LetsEncrypt cert
+package scertecd
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"html"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/tailscale/setec/client/setec"
+	"github.com/tailscale/setec/types/api"
+	"golang.org/x/crypto/acme"
+)
+
+// Server is the scertec updater server.
+//
+// Despite the name "server", it can also be used in a single-shot
+// foreground mode via its UpdateAll method.
```

Comment:
Mention too that one can call UpdateAll without having to Start the server first? (I think that's correct, anyway)